docs/zagg_client_checks.asciidoc

```diff
@@ -0,0 +1,118 @@
+== Writing ZaggClient Checks
+
+This document is intended to be used as a reference to write checks in Openshift V3 using the Zagg interface.
+
+*Note:* This document assumes you are using the https://github.com/openshift/openshift-tools/tree/master/docker/oso-rhel7-zagg-client[oso-rhel7-zagg-client] docker container.  This container includes all the required libraries to communicate with a Zagg server.
+
+==== A Note About ZaggClient Checks
+
+Let's take a quick review and understanding of Zagg and Zabbix to clarify what is meant by 'ZaggClient checks'.
+
+ZaggClient checks are simple gathering of Zabbix items and sending them to the ZaggServer (which eventually sends them to a Zabbix Server).
+
+
+=== Zagg Clients Overview
+
+Currently there are two different methods that can be used to send data to Zagg.
+
+. *ZaggSender python libraries*:  This is the *PREFERRED* way to write checks. ZaggSender is a python interface used to send data to Zagg
+
+. *ops-zagg-client*:  This is a command line tool that can be used to send data to Zagg.
+
+Both of these clients read their default settings from the config file: `/etc/openshift_tools/zagg_client.yaml`.
+
+More information will be provided below on each of these methods
+
+==== ZaggSender - Python API
+
+This is a python API used to create and send data to Zagg.  This is similar to the `zabbix_sender` ruby library used in Openshift V2.
+
+ZaggSender is maintained in the https://github.com/openshift/openshift-tools/tree/master/openshift_tools/monitoring[openshift-tools] repository.  It is installed via the `python-openshift-tools-monitoring` RPM.
+
+ZaggSender reads it's default settings from the config file: `/etc/openshift_tools/zagg_client.yaml`.
+
+The basic bash script would follow the pattern:
+
+.ZaggSender python script basic pattern
+----
+#!/usr/bin/env python
+
+<do stuff, gather info>
+
+zs = ZaggSender()
+zs.add_zabbix_keys({
+                     'item1' : 'value1',
+                     'item2' : 'value2',
+                  })
+zs.send_metrics()
+
+----
+
+Here is an example script of capturing the response code of google.com and sending it to Zagg as Zabbix item "web.google.status":
+
+.ZaggSender Python Example Script
+----
+#!/usr/bin/env python
+
+import urllib
+import yaml
+from openshift_tools.monitoring.zagg_sender import ZaggSender
+
+resp = urllib.urlopen('http://www.google.com')
+google_response = resp.getcode()
+
+zs = ZaggSender()
```

Comment:
Does this now default to grabbing the creds?

I know you say that above, I just didn't know this change was in.

I think I prefer the ctor as it was before, and adding a static method to create the object from the config file. 

Something like this:
```
zs = ZaggSender.from_config()
```

But anyway, if this is already done, then I guess we can just go with it.